src/uwtools/config/formats/yaml.py

```diff
@@ -68,8 +69,12 @@ def _load(self, config_file: Optional[Path]) -> dict:
         with readable(config_file) as f:
             try:
                 config = yaml.load(f.read(), Loader=loader)
-                assert isinstance(config, dict)
-                return config
+                if isinstance(config, dict):
+                    return config
+                raise UWConfigError(
+                    "Parsed a %s value from %s, expected a dict"
+                    % (type(config).__name__, config_file or "stdin")
+                )
```

Comment:
As shown in the PR description, this provides an informative failure message for the user instead of a more obscure stacktrace.